elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (C) 2015 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.stratio.crossdata.connector.elasticsearch
+
+import java.sql.Timestamp
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types._
+import org.elasticsearch.search.SearchHit
+
+object ElasticSearchRowConverter {
+
+  def asRows(schema: StructType, array: Array[SearchHit]): Array[Row] = {
+    import scala.collection.JavaConverters._
+    array map { hit =>
+      hitAsRow(hit.sourceAsMap().asScala.toMap, schema)
+    }
+  }
+
+  def hitAsRow(
+                hits: Map[String, AnyRef],
+                schema: StructType): Row = {
+    val values: Seq[Any] = schema.fields.map {
+      case StructField(name, dataType, _, _) =>
+        hits.get(name).flatMap(v => Option(v)).map(
+          toSQL(_, dataType)).orNull
+    }
+    Row.fromSeq(values)
+  }
+
+  /**
+   * It converts some ES attribute value into
+   * a Row field
+   * @param value Elasticsearch attribute
+   * @param dataType Attribute type
+   * @return The converted value into a Row field.
+   */
+  def toSQL(value: Any, dataType: DataType): Any = {
+    Option(value).map { value =>
+      (value, dataType) match {
+        case _ =>
+          //Assure value is mapped to schema constrained type.
+          enforceCorrectType(value, dataType)
+      }
+    }.orNull
+  }
+
+  protected def enforceCorrectType(value: Any, desiredType: DataType): Any ={
+    if (value == null) {
+      null
+    } else {
+      desiredType match {
+        case StringType => value.toString
+        case _ if value == null || value == "" => null // guard the non string type
```

Comment:
I would move the logic of this line to the line 61 in order to group the cases where null must be returned according to the param value